### DISCUSSION TITLE
Trim only transparent backgrounds?

### DISCUSSION THREAD

**Discussion by zappzerapp** [ImageMagick/ImageMagick#5220] [Help] [answered]
Hi all,

I use trim to cut away transparent backgrounds.
Unfortunately ImageMagick (7.1.0-29) also cuts away solid color backgrounds.

Can I explicitly cut only transparent backgrounds?

I expect input.png and output.png to look visually the same.
Here is my test file:
[trim_issue.zip](https://github.com/ImageMagick/ImageMagick/files/8891226/trim_issue.zip)

And my command:
`convert input.png -trim output.png`

**Comment by fmw42** [comment]
Your input png does not have any transparency. Please post the proper original. 

Or are you saying that you need a solution that does not affect totally opaque input and only processed transparent images.

If so, separated the alpha channel. If it is opaque, then -trim will try to trim it all away and the output will have 0 size or your will get a message.  If it has some transparency, then the trim of the alpha channel will leave only the corresponding opaque parts and you can get the bounding box for that and then just crop the input.

**Comment by snibgo** [comment, marked as answer]
Your input.png has no transparency. If you want an operation that only removes transparent pixels (or pixels of some given colour), a trick is to first *add* a border of colour "none" (or some other color).
```
magick input.png -bordercolor None -border 1 -trim +repage out.png
```

**Comment by zappzerapp** [comment]
Thanks for the quick help 👍 

So there is the PNG with the gray background which should be 2000x2000px in the end and the same image with transparency which should be 2000x1400px

My command cropped both images, so I uploaded only the one without transparency.
